components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/EndpointResource.java

```diff
@@ -87,7 +88,7 @@ public boolean invoke(MessageContext messageContext,
                 if (payload.has(NAME) && payload.has(STATUS)) {
                     changeEndpointStatus(axis2MessageContext, synapseConfiguration, payload);
                 } else {
-                    Utils.setJsonPayLoad(axis2MessageContext, Utils.createJsonErrorObject("Missing parameters in payload"));
+                    handleTracing(payload.get(NAME).getAsString(), messageContext, axis2MessageContext);
```

Comment:
Don't we need to check if the tracing element is present in the payload?

Reply:
What kind of a payload are we expecting here?

Reply:
We handle both updating trace, an erroneous scenario inside handleTracing() method. We check for the tracing element in the common handleTracing() method we have in the Utils.java class. https://github.com/wso2/micro-integrator/blob/master/components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java#L124.

We use the same payload structure as of the one we use to activate/deactivate endpoints.

`curl -X POST https://localhost:9164/management/endpoints  -H 'authorization: Bearer TOKEN' -H 'content-type: application/json' -d '{"name": "endpointName", "trace": "enable"} -k`